libkbfs/block_retrieval_queue.go

```diff
@@ -276,17 +282,22 @@ func (brq *blockRetrievalQueue) Shutdown() {
 }
 
 // TogglePrefetcher allows upstream components to turn the prefetcher on or off
-func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) error {
-	brq.prefetchMtx.Lock()
-	defer brq.prefetchMtx.Unlock()
+func (brq *blockRetrievalQueue) TogglePrefetcher(ctx context.Context, enable bool) (err error) {
+	var shutdownCh <-chan struct{}
+	func() {
+		brq.prefetchMtx.Lock()
+		defer brq.prefetchMtx.Unlock()
+		shutdownCh = brq.prefetcher.Shutdown()
```

Comment:
We never want the prefetcher to be `nil`, we just want it to be shut down, which causes any prefetch requests to no-op.